backends/backends.c

```diff
@@ -1124,7 +1124,11 @@ void *backends_main(void *ptr) {
                 }
             }
             else {
-                error("BACKEND: failed to update database backend '%s'", destination);
+                if (!strstr(destination,"//")) {
+                    error("BACKEND: failed to update database backend '%s'", destination);
+                } else {
+                    error("BACKEND: failed to connect to destination '%s', please remove '//'.", destination);
```

Comment:
it is likely i missing context, but

> `failed to update database backend`

This looks like runtime error, is it? (i think `database` word is not needed btw).


> `failed to connect to destination '%s', please remove '//'.`

This looks like initialization error, is it?

Reply:
The first is a run time error as you described. I kept the word `database`, because it was present in the original message.
And you are also correct about the second, because case the wrong string is present, it won't be able to connect.

Reply:
Why it even try to connect if `dst` is wrong? My question is how come runtime and init errors are live in the same place? I dont ask to fix/do something, just noticed strange thing imo.

Reply:
The goal of this PR is to give messages to clarify what is happening, I agree that to connect again is an error, I did not create any other PR to fix this, because we already had defined that we will have the new `exporter engine` that will be responsible to obsolete this code.
When the user created #4591, there was an error description only for `stream`, but while I was working, I discovered that the error could also happen with `backend`, this is the motive I extended the PR.

Reply:
Well, to be honest your fix is too specific (just check for `//`)

> If you specify the stream destination host like tcp://examle.com netdata tells you that the connect failed, instead of an more appropriate syntax error

The fix should be:
 - introduce `validate_destination` function and validate destination. Report syntax error if syntax is invalid.

Reply:
@amoss what do you think? The problem in `https://github.com/netdata/netdata/issues/4591` is lack of appropriate error message in case of invalid `destination` parameter. So we need to validate it first and log error if it is invalid. 

Do you think current fix is ok?

Reply:
Well... there are two competing concerns here:
1. We have spent far too much time on this issue already, in comparison to the value that it delivers to users and relative to the other work that we have to do.
2. I explained the nice way to fix it back in November.

There are now two possibilities:
1. Fix it quickly
2. Fix it in the cleanest way possible (by removing the use of the alternative separators and documenting the change).

@thiagoftsm / @ilyam8 Which of the two possibilities do you prefer?

Reply:
imo this PR doesnt fix https://github.com/netdata/netdata/issues/4591 and i would suggest to not merge it as it is

https://github.com/netdata/netdata/issues/4591 is low prio, can be fixed later

Reply:
@ilyam8  from #4591 we have the following request:

```
Some kind of better error message, that tells you what the problem is, instead of sending one down the wrong rabbit hole (I was sure it was a network problem until i attached gdb)
```

What am I missing here?
@amoss I think that now we can do the quick fix  and in the future when it is necessary to do a big change inside the daemon (This will happen when we change pthreads to libuv ).

Reply:
Ok, i will try to make it clear.

Current logic: 
 - do not validate destination
 - try to connect
 - if connection failed log `failed to update database backend '%s' destination`

Problem in https://github.com/netdata/netdata/issues/4591 - **lack of syntax validation** of `destination` parameter. Therefore netdata will try to connect to anything.

___

Example: `destination = ?!?!??!?!`

```
2020-01-20 09:51:06: netdata INFO  : BACKENDS : enabled backend for host 'localhost'
2020-01-20 09:51:06: netdata ERROR : BACKENDS : BACKEND: failed to update database backend '?!?!??!?!'
```

> BACKEND: failed to update database backend '?!?!??!?!'

As you see your `check for // in the end` doesnt do.

___

Problem: lack of syntax validation of `destination` parameter.

Soultion: add syntax validation of `destination` parameter.

Logic should be:

 - ~~do not~~ validate destination
 - if destination is invalid, log appropriate message and exit
 - try to connect
 - if connection failed log `failed to update database backend '%s' destination`

Reply:
I don't understand why removing the extra separators and fixing it cleaning requires a major change to the daemon that should be delayed until pthreads are converted to libuv - could you explain why this is?

Reply:
I agree with @ilyam8 and @amoss here. First of all, I also don't see why we need to wait for the libuv transition to fix the input validation. And I think @ilyam8 succinctly presented what the logic should be here.

Reply:
Thanks a lot to express your opinion here.
I changed the code for we connect only when the destination address is valid. When it is not valid we will never connect and we will write only one time the error inside `error.log`.
About to try to reconnect when we fail one time, I am keeping the rule that is already present.